pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/ServerCnx.java

```diff
@@ -138,8 +144,52 @@ public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws E
     // ////
     // // Incoming commands handling
     // ////
+    
+    
+    @Override
+    protected void handleLookup(CommandLookupTopic lookup) {
+        if (log.isDebugEnabled()) {
+            log.debug("Received Lookup from {}", remoteAddress);
+        }
+        lookupDestinationAsync(getBrokerService().pulsar(), DestinationName.get(lookup.getTopic()),
+                lookup.getAuthoritative(), getRole(), lookup.getRequestId()).thenAccept(lookupResponse -> {
+                    ctx.writeAndFlush(lookupResponse);
+                }).exceptionally(ex -> {
+                    // it should never happen
```

Comment:
Actually, ```lookupDestinationAsync(..)``` handles all failure scenarios (auth, no zk session,..) and always returns appropriate ```ByteBuf``` with specific error-response. therefore, future should not complete with exception.

I have addressed all other comments as well.